### PR TITLE
[MM-49088] Catching file upload issue when the response doesn't contain JSON

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
A file upload can get stuck in processing when the error response doesn't contain JSON

#### Ticket Link
Latest comment on https://mattermost.atlassian.net/browse/MM-49088

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by JulienTant — APPROVED** [review]
Thanks @BenCookie95

**Review by pvev — APPROVED** [review]
LGTM! thanks @BenCookie95 :shipit:

File: webapp/channels/src/actions/file_actions.ts [hunk 1]
```diff
@@ -88,10 +88,15 @@ export function uploadFile({file, name, type, rootId, channelId, clientId, onPro
 
                     onSuccess(response, channelId, rootId);
                 } else if (xhr.status >= 400 && xhr.readyState === 4) {
-                    const errorResponse = JSON.parse(xhr.response);
-                    const errorMessage =
+                    let errorMessage = '';
+                    try {
+                        const errorResponse = JSON.parse(xhr.response);
+                        errorMessage =
```
Comment (hanzei) [inline webapp/channels/src/actions/file_actions.ts:94]: This is more of a general question: Why are there cases where the sever returns a non-JSON response? Would it make the webapp code easier and more robust if the server would return a JSON response for xhr requests?

Comment (BenCookie95) [reply webapp/channels/src/actions/file_actions.ts:94]: I'm not exactly sure @hanzei, I could not reproduce this locally and it only seemed to happen on the spinwicks

**Review by lindalumitchell — APPROVED** [review]
LGTM! I tested with the files I had previously used to repro the issue, and no longer saw the issue. I also tried with a different, much larger file, and saw the error message as expected. No related issues found in E2Es.

**Pull request merged**